src/Microsoft.AspNetCore.Mvc.Analyzers/SymbolApiResponseMetadataProvider.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    internal class SymbolApiResponseMetadataProvider
+    {
+        internal static IList<ApiResponseMetadata> GetResponseMetadata(ApiControllerTypeCache typeCache, IMethodSymbol methodSymbol)
+        {
+            var responseMetadataAttributes = methodSymbol.GetAttributes(typeCache.IApiResponseMetadataProvider, inherit: true);
+            var metadataItems = new List<ApiResponseMetadata>();
+            foreach (var attribute in responseMetadataAttributes)
+            {
+                var statusCode = GetStatusCode(attribute);
+                var metadata = new ApiResponseMetadata(statusCode, attribute, convention: null);
+
+                metadataItems.Add(metadata);
+            }
+
+            return metadataItems;
+        }
+
+        private static int GetStatusCode(AttributeData attribute)
+        {
+            const int DefaultStatusCode = 200;
+
+            for (var i = 0; i < attribute.NamedArguments.Length; i++)
+            {
+                var namedArgument = attribute.NamedArguments[i];
```

Comment:
But `StatusCode` is `get` only. I think i'm missing something 😢

Reply:
It has a setter - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ProducesResponseTypeAttribute.cs#L49. This shows up if you have `[ProducesResponseType(StatusCode = 200)]` versus the later one which is `[ProducesResponseType(200)]`

Reply:
Ah! Knew I was missing something 😄